libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -362,6 +364,58 @@ struct NoVariablesInInterfaceChecker: public PostTypeChecker::Checker
 	/// Flag indicating whether we are currently inside a StructDefinition.
 	int m_insideStruct = 0;
 };
+
+struct CircularContractDependencies: public PostTypeChecker::Checker
```

Comment:
I'm not sure if the post type checker is the right place for this. You do not really need to traverse ASTs here - all you need is iterate over all contracts. We already do that in the CompilerStack. Maybe it's better to put it there and move it into a function together with the loop that assigns the call graphs to the annotation?